kconfiglib.py

```diff
@@ -826,6 +826,69 @@ def _load_config(self, filename, replace):
                 if not choice._was_set:
                     choice.unset_value()
 
+    def write_autoconf(self, filename,
+                       header="/* Generated by Kconfiglib (https://github.com/ulfalizer/Kconfiglib) */\n"):
+        r"""
+        Writes out symbol values as a C header file.
+
+        filename:
+          Self-explanatory.
+
+        header (default: "/* Generated by Kconfiglib (https://github.com/ulfalizer/Kconfiglib) */\n"):
+          Text that will be inserted verbatim at the beginning of the file. You
+          would usually want it enclosed in "/* */" to make it a C comment,
+          and include a final terminating newline.
+        """
+        with open(filename, "w") as f:
+
+            # Small optimization
+            write = f.write
+            write(header)
+
+            # Avoid duplicates
+            for sym in self.defined_syms:
+                sym._written = False
+
+            node = self.top_node.list
+            if not node:
+                # Empty configuration
+                return
+
+            while 1:
+                item = node.item
+                if isinstance(item, Symbol):
+                    if not item._written:
+                        if item._write_to_conf:
```

Comment:
There's a subtlety here that causes this to break unless all the symbol values have already been calculated: `_write_to_conf` is calculated when the symbol value is calculated.

If you do `val = item.str_value` before checking `_write_to_conf` and use that, it will work fine. `.str_value` is actually a hidden function call (that uses Python `@property` magic).

There's a related note in a comment in `config_string`, but it might be hard to spot unless you already know that that's a hidden function call too. :)